tests/Plugin/PostBigExtractRequestTest.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+/*
+ * This file is part of the Solarium package.
+ *
+ * For the full copyright and license information, please view the COPYING
+ * file that was distributed with this source code.
+ */
+
+namespace Solarium\Tests\Plugin;
+
+use PHPUnit\Framework\TestCase;
+use Solarium\Core\Client\Adapter\AdapterInterface;
+use Solarium\Core\Client\Client;
+use Solarium\Core\Client\Request;
+use Solarium\Core\Event\PostCreateRequest as PostCreateRequest;
+use Solarium\Plugin\PostBigExtractRequest;
+use Solarium\Tests\Integration\TestClientFactory;
+
+class PostBigExtractRequestTest extends TestCase
+{
+    /**
+     * @var PostBigExtractRequest
+     */
+    protected $plugin;
+
+    /**
+     * @var Client
+     */
+    protected $client;
+
+    /**
+     * @var Solarium\QueryType\Extract\Query
+     */
+    protected $query;
+
+    public function setUp(): void
+    {
+        $this->plugin = new PostBigExtractRequest();
+
+        $this->client = TestClientFactory::createWithCurlAdapter();
+        $this->query = $this->client->createExtract();
+    }
+
+    public function testSetAndGetMaxQueryStringLength()
+    {
+        $this->plugin->setMaxQueryStringLength(512);
+        $this->assertSame(512, $this->plugin->getMaxQueryStringLength());
+    }
+
+    public function testSetAndGetCharset()
+    {
+        $this->plugin->setCharset('iso-8859-1');
+        $this->assertSame('iso-8859-1', $this->plugin->getCharset());
+    }
+
+    public function testPostCreateRequest()
+    {
+        $document = $this->query->createDocument();
+        // create a very long list of literals
+        for ($i = 1; $i <= 100; ++$i) {
+            $field_name = "field_{$i}";
+            $document->$field_name = "value $i";
+        }
+        $this->query->setDocument($document);
+        $this->query->setFile(__FILE__);
+
+        $requestOutput = $this->client->createRequest($this->query);
+        $requestInput = clone $requestOutput;
+        $event = new PostCreateRequest($this->query, $requestOutput);
+        $this->plugin->postCreateRequest($event);
+
+        $this->assertSame(Request::METHOD_POST, $requestOutput->getMethod());
+        $nl_pattern = '(?:\\r\\n|\\r|\\n)';
+        foreach (explode('&', urldecode($requestInput->getQueryString())) as $qs_parameter) {
+            $qs_parameter_arr = explode('=', $qs_parameter);
+            $qs_parameter_name = $qs_parameter_arr[0];
+            $qs_parameter_value = $qs_parameter_arr[1];
+            $pattern = "/^-{2}.*?$nl_pattern+^Content-Disposition: form-data; name=\"".preg_quote($qs_parameter_name)."\"$nl_pattern+Content-Type:.*?$nl_pattern+".preg_quote($qs_parameter_value).'/m';
+            $this->assertMatchesRegularExpression($pattern, $requestOutput->getRawData());
+        }
```

Comment:
… and a more literal test of the resulting raw data.

Reply:
any suggestions to help me out are welcome.

Reply:
I think it would be clearer to test `$requestOutput->getRawData()` against the actual expected output. The only 'unknown' is the hash, but a regex can sort that. A temporary small file would be more readable too.